### PR TITLE
Handle colorscale with transparent colors in surface traces

### LINKED ISSUE #4120: Support custom colorscales with rgba on surface traces
[Demo](https://codepen.io/iamnikosk93/pen/JgwyqG?editors=0010)

As you can see on the colorbar next to the plot, most of the surface should be transparent. Instead, it's colorized according to the rgb values of the rgba colorscale (completely ignoring the alpha).

I don't know if this is the expected behavior, but i guess that it's a bug.

Thanks

### PR DESCRIPTION
**Pull request by archmoj**
Supersedes #4332, and fixes #4120.

@plotly/plotly_js 
cc: @lucapinello

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -90,7 +90,7 @@
     "gl-select-box": "^1.0.4",
     "gl-spikes2d": "^1.0.2",
     "gl-streamtube3d": "^1.4.1",
-    "gl-surface3d": "^1.5.2",
+    "gl-surface3d": "git://github.com/gl-vis/gl-surface3d.git#77c2db23f81ba994047f82705afcfcc0d897e154",
```
Comment (archmoj) [inline package.json:93]: https://github.com/gl-vis/gl-surface3d/pull/18

File: src/traces/surface/convert.js [hunk 2]
```diff
@@ -434,7 +434,7 @@ proto.update = function(data) {
     var sceneLayout = scene.fullSceneLayout;
     var surface = this.surface;
     var alpha = data.opacity;
-    var colormap = parseColorScale(data, alpha);
+    var colormap = parseColorScale(data);
```
Comment (alexcjohnson) [inline src/traces/surface/convert.js:437]: Does `trace.opacity` still do anything with these modifications? I'd expect it to effectively multiply the opacity of the entire colorscale.

And do we need any adjustments down here:
https://github.com/plotly/plotly.js/blob/13324e220e22d1f80cb386c4df5af894b18e3996/src/traces/surface/convert.js#L683-L686
?

Comment (archmoj) [reply src/traces/surface/convert.js:436]: > Does `trace.opacity` still do anything with these modifications? I'd expect it to effectively multiply the opacity of the entire colorscale.

Positive. We have mocks for that. Also tested interactively.

Comment (archmoj) [reply src/traces/surface/convert.js:436]: > And do we need any adjustments down here:
> https://github.com/plotly/plotly.js/blob/13324e220e22d1f80cb386c4df5af894b18e3996/src/traces/surface/convert.js#L683-L686
> 
> ?

Good call. Adjustments in https://github.com/plotly/plotly.js/pull/5166/commits/cfe9dd1fd7f1712eaab217210b5a665e4a7fbd50.

So this PR should be ready to go.

**Review by alexcjohnson — APPROVED** [review]
Very nice! 💃 once we have a published `gl-surface3d` to include.

**Pull request merged**